### PR TITLE
Add support for KNXnet/IP Server discovery

### PR DESCRIPTION
**Pull request by stffabi**

### CODE REVIEW

File: knx/knxnet/search.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+	HostInfo
+}
+
+// Service returns the service identifier for search request
+func (SearchReq) Service() ServiceID {
+	return SearchReqService
+}
+
+// SetHostInfo sets the address where ObjectServers should send the reponse to
+func (req *SearchReq) SetHostInfo(addr net.Addr) error {
```
Comment (vapourismo) [inline knx/knxnet/search.go]: Could this be a function `NewHostInfo` returning `(*SearchReq, error)` instead?

Comment (stffabi) [reply knx/knxnet/search.go]: Of course, I can change that.

**Comment by stffabi** [comment]
I've used the specification which could be downloaded [here](https://my.knx.org/en/shop/knx-specifications). It's a zip file with a bunch of PDFs. The relevant PDFs are
- 03_08_02 Core v01.05.01 AS.pdf - 
- [KNX_BAOS_Protocol.pdf](https://weinzierl.de/images/download/products/770/KNX_BAOS_Protocol.pdf) - 4.1. KNXnet/IP discovery algorithm

File: knx/knxnet/search.go (hunk 1 above)
Comment (stffabi) [reply knx/knxnet/search.go]: Done

**Comment by stffabi** [comment]
Awesome, thanks so much for the merge...

**Pull request merged**